core/b2b/client.ts

```diff
@@ -29,9 +32,11 @@ const B2BTokenResponseSchema = z.object({
 });
 
 export async function loginWithB2B({ customerId, customerAccessToken }: LoginWithB2BParams) {
-  const { B2B_API_HOST, B2B_API_TOKEN, BIGCOMMERCE_CHANNEL_ID } = ENV.parse(process);
+  const { B2B_API_HOST, B2B_API_TOKEN, BIGCOMMERCE_CHANNEL_ID, NODE_ENV } = ENV.parse(process);
+
+  const apiHost = NODE_ENV === 'production' ? PROD_B2B_URL : B2B_API_HOST;
```

Comment:
It might be worth moving all this (including the warnings) into their own file:

```tsx
// ./getAPIHostname.ts

const DEFAULT_API_HOST = 'https://api.example.com';

const ENV = z.object({
  env: z.object({
    API_HOST: z.string().default(DEFAULT_API_HOST),
    NODE_ENV: z.enum(['development', 'production']).default('production'),
  })
})
.transform(({ env }) => env);

export const getAPIHostname = (): string => {}
```

Keeping the warn/use and everything in one place makes it easier to understand the reasons for it. Also, not a huge deal to parse `process` twice 🤷 

```tsx
const apiHost = getAPIHostname();
```